### PR TITLE
EmployeeIdentity signing means validation

### PR DESCRIPTION
**Pull request by woutslakhorst**
part of #2047 

Adds validation for presentations created with the employee identity

### CODE REVIEW

File: auth/services/selfsigned/types.go [hunk 1]
```diff
@@ -141,3 +145,39 @@ func (s session) credentialSubject() []interface{} {
...
+}
+
+type employeeIdentityCredentialMember struct {
+	Identifier string                                 `json:"identifier"`
+	Member     employeeIdentityCredentialMemberMember `json:"member"`
+	RoleName   string                                 `json:"roleName"`
+	Type       string                                 `json:"type"`
+}
+
+type employeeIdentityCredentialMemberMember struct {
```
Comment (reinkrul) [inline auth/services/selfsigned/types.go:162]: too many members?

File: auth/services/selfsigned/validator.go [hunk 2]
```diff
@@ -19,42 +19,170 @@
...
+		return
+	}
+	vc := vcs[0]
+	signingMethod := proof.VerificationMethod
+	signingMethod.Fragment = ""
+	if vc.Issuer.String() != signingMethod.String() {
+		resultErr = newVerificationError("signer must be credential issuer")
+		return
+	}
+	bytes, _ = json.Marshal(vc.CredentialSubject[0])
```
Comment (reinkrul) [inline auth/services/selfsigned/validator.go:122]: I think you can use `vc.UnmarshalSubject()`

File: auth/services/selfsigned/validator.go [hunk 3]
```diff
@@ -19,42 +19,170 @@
...
+	if vc.Issuer.String() != credentialSubject.ID {
+		resultErr = newVerificationError("signer must be credentialSubject")
+		return
+	}
+
+	return credentialSubject, proof, nil
+}
+
+func validateRequiredAttributes(credentialSubject employeeIdentityCredentialSubject) error {
+	// check for mandatory attrs
```
Comment (reinkrul) [inline auth/services/selfsigned/validator.go:133]: shouldn't these be validated on the VC already?

**Review by reinkrul — CHANGES_REQUESTED** [review]

File: auth/services/selfsigned/types.go (hunk 1 above)
Comment (woutslakhorst) [reply auth/services/selfsigned/types.go:162]: Unfortunate result of the VC model

File: auth/services/selfsigned/validator.go (hunk 3 above)
Comment (woutslakhorst) [reply auth/services/selfsigned/validator.go:133]: These are the requirements from the spec (rfc that has to be written). It doesn't validate stuff like issuer, context and id but mostly `credentialSubject` requirements.

Or do you mean it should be under vcr/credential/validator? << where credentials are validated that enter the server wallet from the network?

File: auth/services/selfsigned/validator.go (hunk 2 above)
Comment (woutslakhorst) [reply auth/services/selfsigned/validator.go]: Also added check for exactly 1 credentialSubject

File: auth/services/selfsigned/validator.go (hunk 3 above)
Comment (reinkrul) [reply auth/services/selfsigned/validator.go:138]: The last one, indeed. I could've been more clear.

Comment (woutslakhorst) [reply auth/services/selfsigned/validator.go:138]: So we validate identity means (and VPs) inside auth. The attestations (VCs over network) are validated within VCR.

**Review by reinkrul — APPROVED** [review]

**Pull request merged**